### PR TITLE
release 0.33.7

### PR DESCRIPTION
**Pull request by erikgrinaker**
Adds changelog and version bump to release branch

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## v0.33.7
+
+*August 4, 2020*
+
+### BUG FIXES:
+
+- [go] Build release binary using Go 1.14.4, to avoid halt caused by Go 1.14.1 (https://github.com/golang/go/issues/38223)
```
Comment (tessr) [inline CHANGELOG.md:9]: was there a code change for this too anywhere?

Comment (erikgrinaker) [reply CHANGELOG.md:9]: It was a change to the Docker image used for builds, which lives in a separate repo. Want me to dig it up and include a link to it?

Comment (tessr) [reply CHANGELOG.md:9]: nah, that's fine. just thought it would be good to include a PR link if there was one.

Comment (marbar3778) [reply CHANGELOG.md:9]: Tendermint/images. There were no code changes. I just reran the build since we don't specify versions for golang.

Comment (erikgrinaker) [reply CHANGELOG.md:9]: Looks like the build image change was a noop as well, since we're using the `golang:latest` image.

**Pull request merged**